external/storm-solr/README.md

```diff
@@ -0,0 +1,188 @@
+
+
+### SolrJsonMapper
+The `SolrJsonMapper` creates a Solr update request that is sent to the URL endpoint defined by Solr as the resource 
+destination for requests in JSON format. 
+ 
+To create a `SolrJsonMapper` the client must specify the name of the collection to update as well as the 
+tuple field that contains the JSON object used to update the Solr index. If the tuple does not contain the field specified, 
+a `SolrMapperException` will be thrown when the method `toSolrRequest`is called. If the field exists, its value can either 
+be a String with the contents in JSON format, or a Java object that will be serialized to JSON
+ 
+Code snippet illustrating how to create a `SolrJsonMapper` object to update the `gettingstarted` Solr collection with JSON content 
+declared in the tuple field with name "JSON"
+``` java
+    SolrMapper solrMapper = new SolrJsonMapper.Builder("gettingstarted", "JSON").build();
+```
+
+
+### SolrFieldsMapper
+The `SolrFieldsMapper` creates a Solr update request that is sent to the Solr URL endpoint that handles the updates of `SolrInputDocument` objects.
+
+To create a `SolrFieldsMapper` the client must specify the name of the collection to update as well as the `SolrSchemaBuilder`. 
+The Solr `Schema` is used to extract information about the Solr schema fields and corresponding types. This metadata is used
+to get the information from the tuples. Only tuple fields that match a static or dynamic Solr fields are added to the document. Tuple fields 
+that do not match the schema are not added to the `SolrInputDocument` being prepared for indexing. A debug log message is printed for the 
+ tuple fields that do not match the schema and hence are not indexed.
+ 
+
+The `SolrFieldsMapper` supports multivalue fields. A multivalue tuple field must be tokenized. The default token is |. Any 
+arbitrary token can be specified by calling the method `org.apache.storm.solr.mapper.SolrFieldsMapper.Builder.setMultiValueFieldToken`
+that is part of the `SolrFieldsMapper.Builder` builder class. 
+
+Code snippet illustrating how to create a `SolrFieldsMapper` object to update the `gettingstarted` Solr collection. The multivalue 
+field separates each value with the token % instead of the default | . To use the default token you can ommit the call to the method
+`setMultiValueFieldToken`.
+
+``` java
+    new SolrFieldsMapper.Builder(
+            new RestJsonSchemaBuilder("localhost", "8983", `gettingstarted`), `gettingstarted`)
+                .setMultiValueFieldToken("%").build();
+```
+
+# Build And Run Bundled Examples  
+To be able to run the examples you must first build the java code in the package `storm-solr`, 
+and then generate an uber jar with all the dependencies.
+
+
+## Build the Storm Apache Solr Integration Code
+
+`mvn clean install -f REPO_HOME/storm/external/storm-solr`
+ 
+## Use the Maven Assembly Plugin to Build the Uber Jar
```

Comment:
We should encourage the use of the Maven shade plugin over the assembly plugin. The assembly plugin doesn't handle jar manifest data properly and can cause problems with Hadoop dependencies.